dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/programrule/implementers/enrollment/ShowErrorWarningExecutorTest.java

```diff
@@ -0,0 +1,173 @@
+        getErrorActionRule() );
+
+    private ShowErrorExecutor showErrorExecutor = new ShowErrorExecutor( getErrorActionRule() );
+
+    private ShowWarningExecutor showWarningExecutor = new ShowWarningExecutor( getErrorActionRule() );
+
+    private TrackerBundle bundle;
+
+    @Mock
+    private TrackerPreheat preheat;
+
+    @BeforeEach
+    void setUpTest()
+    {
+        bundle = TrackerBundle.builder().build();
+        bundle.setEnrollments( getEnrollments() );
+        bundle.setPreheat( preheat );
+    }
+
+    @Test
+    void shouldReturnAnErrorWhenAShowErrorActionIsTriggeredForActiveEnrollment()
+    {
+        Optional<ProgramRuleIssue> error = showErrorExecutor.executeEnrollmentRuleAction( bundle, activeEnrollment() );
+        assertTrue( error.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAnErrorWhenAShowErrorActionIsTriggeredForCompletedEnrollment()
+    {
+        Optional<ProgramRuleIssue> error = showErrorExecutor.executeEnrollmentRuleAction( bundle,
+            completedEnrollment() );
+        assertTrue( error.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAWarningWhenAShowErrorActionIsTriggeredForActiveEnrollment()
+    {
+        Optional<ProgramRuleIssue> warning = showWarningExecutor.executeEnrollmentRuleAction( bundle,
+            activeEnrollment() );
+        assertTrue( warning.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAWarningWhenAShowErrorActionIsTriggeredForCompletedEnrollment()
+    {
+        Optional<ProgramRuleIssue> warning = showWarningExecutor.executeEnrollmentRuleAction( bundle,
+            completedEnrollment() );
+        assertTrue( warning.isPresent() );
+    }
+
+    @Test
+    void shouldNotReturnAnErrorWhenAShowErrorOnCompleteActionIsTriggeredForActiveEnrollment()
+    {
+        Optional<ProgramRuleIssue> error = errorOnCompleteExecutor.executeEnrollmentRuleAction( bundle,
+            activeEnrollment() );
+        assertFalse( error.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAnErrorWhenAShowErrorOnCompleteActionIsTriggeredForCompletedEnrollment()
+    {
+        Optional<ProgramRuleIssue> error = errorOnCompleteExecutor.executeEnrollmentRuleAction( bundle,
+            completedEnrollment() );
+        assertTrue( error.isPresent() );
+    }
+
+    @Test
+    void shouldNotReturnAWarningWhenAShowErrorOnCompleteActionIsTriggeredForActiveEnrollment()
+    {
+        Optional<ProgramRuleIssue> warning = warningOnCompleteExecutor.executeEnrollmentRuleAction( bundle,
+            activeEnrollment() );
+        assertFalse( warning.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAWarningWhenAShowErrorOnCompleteActionIsTriggeredForCompletedEnrollment()
+    {
+        Optional<ProgramRuleIssue> warning = warningOnCompleteExecutor.executeEnrollmentRuleAction( bundle,
+            completedEnrollment() );
+        assertTrue( warning.isPresent() );
```

Comment:
Do we expect anything in particular about the warnings/errors (ProgramRuleIssue) that the executor returns? I wonder if we should assert anything on it.